tests/test_deployments.py

```diff
@@ -1057,6 +1071,26 @@ async def test_run_deployment_with_deployment_id_uuid(
         assert flow_run.deployment_id == deployment_id
         assert flow_run.state
 
+    async def test_run_deployment_emits_warning(
```

Comment:
Can you add a check to ensure the provided job variables show up as expected on the created flow run? That'll ensure a test fails if the plumbing from `run_deployment` through the client is broken.